semeio/jobs/auto_correlate_observations/config.py

```diff
@@ -0,0 +1,315 @@
+            cs.MetaKeys.Description: "TODO",
+        },
+        _ALPHA: {cs.MetaKeys.Type: cs.types.Number, cs.MetaKeys.Description: "TODO",},
+    },
+}
+
+
+class _BooleanWithMessage:
+    def __init__(self, value, msg):
+        self._value = value
+        self._msg = msg
+
+    def __nonzero__(self):
+        return self._value is True
+
+    def __bool__(self):
+        return self._value is True
+
+    def __and__(self, other):
+        return bool(self) and bool(other)
+
+    @property
+    def msg(self):
+        return self._msg
+
+
+# NB: This is a validator, not a transformation.
+# This is a hack to avoid configsuite wrapping the return value
+@cs.transformation_msg("Ensures that all requested observations are indeed present")
+def _observations_present(observation_key, context):
+    if observation_key in context.observation_keys:
+        return _BooleanWithMessage(
+            True, "Observation {} found".format(observation_key),
+        )
+    else:
+        return _BooleanWithMessage(
+            False, "Found no match for observation {}".format(observation_key),
+        )
+
+
+@cs.transformation_msg("Inject all as default and expand filters")
+def _realize_filters(observation_keys, context):
+    all_keys = tuple(context.observation_keys)
+    if len(observation_keys) == 0:
+        observation_keys = ("*",)
+
+    matches = set()
+    for obs_filter in observation_keys:
+        new_matches = set(fnmatch.filter(all_keys, obs_filter))
+        if len(new_matches) == 0:
+            new_matches = set((obs_filter,))
+        matches = matches.union(new_matches)
+
+    return tuple(matches)
+
+
+_OBSERVATION_SCHEMA = {
+    cs.MetaKeys.Type: cs.types.List,
+    cs.MetaKeys.Description: "TODO",
+    cs.MetaKeys.ContextTransformation: _realize_filters,
+    cs.MetaKeys.Content: {
+        cs.MetaKeys.Item: {
+            cs.MetaKeys.Type: cs.types.String,
+            cs.MetaKeys.ContextValidators: (_observations_present,),
+        }
+    },
+}
+
+
+_SCHEMA = {
+    cs.MetaKeys.Type: cs.types.NamedDict,
+    cs.MetaKeys.Description: "TODO",
+    cs.MetaKeys.Content: {
+        _OBSERVATIONS: _OBSERVATION_SCHEMA,
+        _CLUSTERING: _CLUSTERING_SCHEMA,
+        _SCALING: _SCALING_SCHEMA,
+    },
+}
+
+
+_DEFAULT_VALUES = {
+    _SCALING: {_THRESHOLD: 0.95, _STD_CUTOFF: 1e-6, _ALPHA: 3.0,},
+    _OBSERVATIONS: (),
+    _CLUSTERING: {
+        _METHOD: _SPEARMAN_CORRELATION,
+        _SPEARMAN_CORRELATION: {
+            _FCLUSTER: {_CRITERION: _INCONSISTENT, _DEPTH: 2,},
+            _LINKAGE: {_METHOD: _SINGLE, _METRIC: _EUCLIDEAN,},
+        },
+    },
+}
+
+
+class _ObservationContext:
+    def __init__(self, observation_keys):
+        Context = collections.namedtuple("Context", ("observation_keys",))
+        self._context = Context(set(observation_keys))
+
+    def __call__(self, _):
+        return self._context
+
+
+# TODO: Should we rename this one?
+class AutoCorrelateConfig(cs.ConfigSuite):
```

Comment:
Hm and why not `AutoScalingConfig`?

Reply:
I think we should change the name away from `Auto` and then consistently change these names.